dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ConfigurationRestRepository.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import javax.annotation.Resource;
+
+import org.dspace.app.rest.exception.RepositoryMethodNotImplementedException;
+import org.dspace.app.rest.model.PropertyRest;
+import org.dspace.core.Context;
+import org.dspace.services.ConfigurationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.stereotype.Component;
+
+/**
+ * This is the repository responsible of exposing configuration properties
+ */
+@Component(PropertyRest.CATEGORY + "." + PropertyRest.NAME)
+public class ConfigurationRestRepository extends DSpaceRestRepository<PropertyRest, String> {
+    @Autowired
+    private ConfigurationService configurationService;
+
+    @Resource(name = "exposedConfigurationProperties")
+    private ArrayList<String> exposedProperties;
+
+    /**
+     * Gets the value of a configuration property if it is exposed via REST
+     *
+     * Example:
+     * <pre>
+     * {@code
+     * curl http://<dspace.server.url>/api/config/properties/google.analytics.key
+     *  -XGET \
+     *  -H 'Authorization: Bearer eyJhbGciOiJI...'
+     * }
+     * </pre>
+     *
+     * @param property
+     * @return
+     */
+    @Override
+    public PropertyRest findOne(Context context, String property) {
+        if (!exposedProperties.contains(property) || configurationService.getArrayProperty(property).length == 0) {
```

Comment:
You can replace this `getArrayProperty().length` check with a call to `configurationService.hasProperty()`.  That's the recommended way to determine if a property exists in our configuration.